[BUG]: options did not refresh when empty array proxy was updated

toranb opened the pull request:
I was able to get a solid test to show the production bug Scott and I found recently ( mentioned in detail using issue #142 )

I could use a little feedback on "where this fits" in the integration test file (to make sure you are happy with it/etc). I also updated the doc string you had above the block (again -unsure if I did this as you expected).

Let me know what I can do to pull this in/ get a beta 4 some time this week :)

cibernox commented:
That was much simpler than I thought. 

About updating the comments, I'm sure they are not completely up to date anyway. Before implementing the test suite I spent a couple hours gathering all the possible scenarios to tests and wrote comments for them before implementing them. Once they are implemented they are not really necessary.

Can you only add another test to check that it also works when updating pushing to an array that is not empty?

I'll push a new beta tomorrow.

Thanks! :bow:

toranb commented:
Sounds good! I went ahead and removed the comments I made at the top (not needed as you mentioned). I added the test you asked about (maybe it's worth mentioning -it didn't require any production code beyond what I added for the first -a valid regression test if nothing else).

If you need anything else let me know :)

cibernox commented:
Nope, this is perfect.

Thanks again!
